packages/web_benchmarks/lib/analysis.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'server.dart';
+
+export 'src/benchmark_result.dart';
+
+/// Returns the average of the benchmark results in [results].
+///
+/// Each element in [results] is expected to have identical benchmark names and
+/// metrics; otherwise, an [Exception] will be thrown.
+BenchmarkResults computeAverage(List<BenchmarkResults> results) {
+  if (results.isEmpty) {
+    throw Exception('Cannot take average of empty list.');
+  }
+
+  BenchmarkResults totalSum = results.first;
+  for (int i = 1; i < results.length; i++) {
+    final BenchmarkResults current = results[i];
+    totalSum = totalSum._sumWith(current);
+  }
+
+  final Map<String, List<Map<String, Object?>>> average = totalSum.toJson();
```

Comment:
It's interesting that you had to pipe it through JSON. It tells me something is missing. Can we extend the API of `BenchmarkResults` so you can do the same computation without JSON?